Classes/Core/ConsoleApplication.php

```diff
@@ -0,0 +1,252 @@
+<?php
+namespace Helhum\Typo3Console\Core;
+
+/*
+ * This file is part of the TYPO3 Console project.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ *
+ * For the full copyright and license information, please read
+ * LICENSE file that was distributed with this source code.
+ *
+ */
+
+use Helhum\Typo3Console\Core\Booting\RunLevel;
+use Helhum\Typo3Console\Error\ExceptionHandler;
+use Helhum\Typo3Console\Mvc\Cli\RequestHandler;
+use Symfony\Component\Console\Input\ArgvInput;
+use TYPO3\CMS\Core\Core\ApplicationInterface;
+use TYPO3\CMS\Core\Package\PackageManager;
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Mvc\Cli\CommandManager;
+use TYPO3\CMS\Extbase\Mvc\Cli\Response;
+
+/**
+ * @internal
+ */
+class ConsoleApplication implements ApplicationInterface
+{
+    /**
+     * @var ConsoleBootstrap
+     */
+    private $bootstrap;
+
+    public function __construct(\Composer\Autoload\ClassLoader $classLoader)
+    {
+        $this->ensureRequiredEnvironment();
+        $this->bootstrap = ConsoleBootstrap::getInstance();
+        $this->bootstrap->initializeClassLoader($classLoader);
+    }
+
+    /**
+     * Bootstraps the minimal infrastructure, registers a request handler and
+     * then passes control over to that request handler.
+     *
+     * @param callable|null $execute
+     */
+    public function run(callable $execute = null)
+    {
+        $this->boot();
+
+        $this->bootstrap->registerRequestHandlerImplementation(RequestHandler::class);
+        $this->bootstrap->handleRequest(new ArgvInput());
+
+        if ($execute !== null) {
+            call_user_func($execute);
+        }
+
+        $this->shutdown();
+    }
+
+    /**
+     * Checks PHP sapi type and sets required PHP options
+     */
+    private function ensureRequiredEnvironment()
+    {
+        if (PHP_SAPI !== 'cli' || !isset($_SERVER['argc'], $_SERVER['argv'])) {
```

Comment:
I think `phpdbg` should be supported too.

Reply:
That is not new. If so, this should be done in a separate pr